rules/aws_resource_missing_tags.go

```diff
@@ -70,36 +144,79 @@ func (r *AwsResourceMissingTagsRule) Check(runner tflint.Runner) error {
 		// Special handling for tags on aws_autoscaling_group resources
 		if resourceType == "aws_autoscaling_group" {
 			err := r.checkAwsAutoScalingGroups(runner, config)
+			err = runner.EnsureNoError(err, func() error {
+				return nil
+			})
 			if err != nil {
 				return err
 			}
 			continue
 		}
 
 		resources, err := runner.GetResourceContent(resourceType, &hclext.BodySchema{
-			Attributes: []hclext.AttributeSchema{{Name: tagsAttributeName}},
+			Attributes: []hclext.AttributeSchema{
+				{Name: tagsAttributeName},
+				{Name: providerAttributeName},
+			},
 		}, nil)
 		if err != nil {
 			return err
 		}
 
+		if resources.IsEmpty() {
+			continue
+		}
+
 		for _, resource := range resources.Blocks {
-			if attribute, ok := resource.Body.Attributes[tagsAttributeName]; ok {
-				logger.Debug("Walk `%s` attribute", resource.Labels[0]+"."+resource.Labels[1]+"."+tagsAttributeName)
+			providerName := "default"
+			if val, ok := resource.Body.Attributes[providerAttributeName]; ok {
+				err := runner.EvaluateExpr(
+					val.Expr,
+					&providerName,
+					nil,
+				)
```

Comment:
The `provider` meta-argument is not an evaluable expression.
https://developer.hashicorp.com/terraform/language/providers/configuration#selecting-alternate-provider-configurations

A dedicated decoding function is needed to handle this. See also discussion in https://github.com/terraform-linters/tflint-ruleset-aws/pull/342#issuecomment-1118607442.
Perhaps we should generalize this API so that it can also be referenced in this rule, or add a new API to Runner.

Reply:
Fixed

Reply:
I'll happily add logic to the runner api to decode provider tags if needed in a separate PR :)